src/providers/token/token.ts

```diff
@@ -23,4 +25,41 @@ export class TokenProvider {
     });
   }
 
+  /**
+   * Function to set listener on default tokens contracts
+   * @param address wallet address
+   * @param tokens Array of token object
+   * @param provider ether provider
+   */
+  public setTokenListener(
+    address: string,
+    tokens: Array<{
+      contractAddress: string,
+      name: string,
+      symbol: string,
+      decimals: number
+    }>,
+    provider: any
+  ) {
+    for(let i=0; i<tokens.length; i++) {
+      //test notification
+      this.notificationProvider.scheduleNotification('Single ILocalNotification', 'test');
+
+      let contract =  new ethers.Contract(tokens[i].contractAddress, erc20Abi, provider);
+      console.log(contract);
+      //filter for Transfer event only
+      let filter = contract.filters.Transfer(null, address);
+
+      //check that there is no already a listener on that contract event
+      if(contract.listenerCount() == 0) {
+        contract.on(filter, (from, to, value) => {
+          console.log('I received ' + value.toString() + ' tokens from ' + from);
+          let text = `Received ${value.toString} ${tokens[i].name} from ${from}`;
+          let data = 'test';
+          this.notificationProvider.scheduleNotification(text, data);
+        });
+      }
+    }
+  }
+
```

Comment:
@bumi  would love to get ur feedback about this function whenever ur are available.
Here we pass array of token Object and initiate new ethers.Contract() to each token contract and setup a listener.

That should work well in a for loop right ?